### PR TITLE
Signup: Fix Themes step horizontal scroll

### PR DESCRIPTION
**Pull request by bisko**
This PR is a fix for #8745.

The Pressable "substeps" in Signup break the design a bit by being shifted to the side. Even though they are hidden, they still expand the bounding box and cause the scroll.

I'm trying to make the substeps play nice with the main steps and not cause such problems.

To test:
* Start Signup
* Reach Themes step
* Check if there is horizontal scroll - there shouldn't be
* Check if the `Upload theme` option is working correctly
* Check the Themes step submits properly
* Finish Signup and verify that the resulting site has the chosen theme
* Verify no JS errors appear in the console

Note:
After starting signup, you need to get yourself in the Theme Upload AB test, as @lsinger mentioned below:
> FWIW, to activate the Upload Theme option:
>
> `localStorage.setItem('ABTests', '{"signupThemeUpload_20160928": "showThemeUpload"}')`

### CODE REVIEW

**Comment by bisko** [comment]
[Issue #8745 Signup: Theme step has horizontal scrollbar in Firefox](https://trello.com/c/ZEVKv183/39-issue-8745-signup-theme-step-has-horizontal-scrollbar-in-firefox)

**Comment by lsinger** [comment]
FWIW, to activate the `Upload Theme` option:

`localStorage.setItem('ABTests', '{"signupThemeUpload_20160928": "showThemeUpload"}')`

File: client/signup/steps/theme-selection/index.jsx [hunk 1]
```diff
@@ -101,42 +123,27 @@ class ThemeSelectionStep extends Component {
...
-		const pressableWrapperClassName = classNames( {
-			'theme-selection__pressable-wrapper': true,
-			'is-hidden': ! this.state.showPressable,
-		} );
-
-		const themesWrapperClassName = classNames( {
-			'theme-selection__themes-wrapper': true,
-			'is-hidden': this.state.showPressable,
-		} );
+		let headerText = translate( 'Choose a theme.' );
```
Comment (lsinger) [inline client/signup/steps/theme-selection/index.jsx]: I think we'd prefer to avoid the use of `let` in favor of `const` if possible. 

In this case:

```
const headerText = this.state.showPressable ? "...?" : "...!";
// etc.
```

Comment (bisko) [inline client/signup/steps/theme-selection/index.jsx]: I did it with the shorthand `if` syntax and `const`, but didn't look very nice and wasn't very obvious what was happening there:
```
const headerText = this.state.showPressable
	? translate( 'Upload your WordPress Theme' )
	: translate( 'Choose a theme.' );

const subHeaderText = this.state.showPressable
	? translate( 'Our partner Pressable is here to help you' )
	: translate( 'No need to overthink it. You can always switch to a different theme later.' );
```

versus:

```
let headerText = translate( 'Choose a theme.' );
let subHeaderText = translate( 'No need to overthink it. You can always switch to a different theme later.' );

if ( this.state.showPressable ) {
	headerText = translate( 'Upload your WordPress Theme' );
	subHeaderText = translate( 'Our partner Pressable is here to help you' );
}
```

I think the second one is better for readability if you're quickly skimming through the code and I don't think `let`/`const` would make a huge difference in terms of performance here :)

File: client/signup/step-wrapper/index.jsx [hunk 2]
```diff
@@ -14,7 +14,14 @@ import config from 'config';
 export default React.createClass( {
 	displayName: 'StepWrapper',
 
+	propTypes: {
+		hideNavButtons: React.PropTypes.bool
```
Comment (mcsf) [inline client/signup/step-wrapper/index.jsx]: Drive-by comment: `shouldHideNav[Buttons]`

File: client/signup/steps/theme-selection/index.jsx [hunk 3]
```diff
@@ -66,13 +66,34 @@ class ThemeSelectionStep extends Component {
 	};
 
 	renderThemesList() {
+		const pressableWrapperClassName = classNames( {
+			'theme-selection__pressable-wrapper': true,
```
Comment (mcsf) [inline client/signup/steps/theme-selection/index.jsx]: If this key is always true I think classnames allows single strings as arguments. Also, check which is more common in calypso: classnames or classNames

File: client/signup/steps/theme-selection/index.jsx [hunk 4]
```diff
@@ -66,13 +66,34 @@ class ThemeSelectionStep extends Component {
...
+			'is-hidden': ! this.state.showPressable,
+		} );
+
+		const themesWrapperClassName = classNames( {
+			'theme-selection__themes-wrapper': true,
```
Comment (mcsf) [inline client/signup/steps/theme-selection/index.jsx]: Same applies here

File: client/signup/steps/theme-selection/index.jsx [hunk 5]
```diff
@@ -66,13 +66,34 @@ class ThemeSelectionStep extends Component {
...
-			<SignupThemesList
-				surveyQuestion={ this.props.chosenSurveyVertical }
-				designType={ this.props.designType || this.props.signupDependencies.designType }
-				handleScreenshotClick={ this.pickTheme }
-				handleThemeUpload={ this.handleThemeUpload }
-			/>
+			<div className="theme-selection__pressable-substep-wrapper">
+				<div className={ pressableWrapperClassName }>
+					<PressableThemeStep
+						{ ... this.props }
```
Comment (mcsf) [inline client/signup/steps/theme-selection/index.jsx]: Remove space after ellipsis

**Comment by lsinger** [comment]
Thanks for the update! 👍 :shipit:

File: client/signup/steps/theme-selection/index.jsx [hunk 6]
```diff
@@ -101,42 +122,27 @@ class ThemeSelectionStep extends Component {
...
-			'is-hidden': this.state.showPressable,
-		} );
+		const headerText = this.state.showPressable
+			? translate( 'Upload your WordPress Theme' )
+			: translate( 'Choose a theme.' );
 
-		const { translate } = this.props;
+		const subHeaderText = this.state.showPressable
+			? translate( 'Our partner Pressable is here to help you' )
+			: translate( 'No need to overthink it. You can always switch to a different theme later.' );
```
Comment (mcsf) [inline client/signup/steps/theme-selection/index.jsx]: Maybe add context for translators. The other strings may not need it, but this one probably does.

File: client/signup/steps/theme-selection/index.jsx (hunk 3 above)
Comment (bisko) [inline client/signup/steps/theme-selection/index.jsx]:
```
633 classNames
573 classnames
```
Take your pick :) I think `classNames`, since this is also the way it is named in the official package documentation.

File: client/signup/steps/theme-selection/index.jsx [hunk 7]
```diff
@@ -101,42 +122,31 @@ class ThemeSelectionStep extends Component {
...
+		const headerText = this.state.showPressable
+			? translate( 'Upload your WordPress Theme' )
+			: translate( 'Choose a theme.' );
 
-		const { translate } = this.props;
+		const subHeaderText = this.state.showPressable
+			? translate( 'Our partner Pressable is here to help you', {
+				context: 'Subheader text in Signup, when a user chooses the Upload theme in the Themes step'
+			} )
+			: translate( 'No need to overthink it. You can always switch to a different theme later.', {
```
Comment (a8ci18n) [inline client/signup/steps/theme-selection/index.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'No need to overthink it. You can always switch to a different theme later.' )` *ES Score: 14.48*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/signup-themes-step-pressable-overlay-no-scroll)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/theme-selection/index.jsx [hunk 8]
```diff
@@ -101,42 +122,31 @@ class ThemeSelectionStep extends Component {
...
-			'theme-selection__themes-wrapper': true,
-			'is-hidden': this.state.showPressable,
-		} );
+		const headerText = this.state.showPressable
+			? translate( 'Upload your WordPress Theme' )
+			: translate( 'Choose a theme.' );
 
-		const { translate } = this.props;
+		const subHeaderText = this.state.showPressable
+			? translate( 'Our partner Pressable is here to help you', {
```
Comment (a8ci18n) [inline client/signup/steps/theme-selection/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Our partner Pressable is here to help you' )` *ES Score: 13.91*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/signup-themes-step-pressable-overlay-no-scroll)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).
